### PR TITLE
bpo-24165: Add a freelist for long objects

### PR DESCRIPTION
**Pull request by methane**
https://bugs.python.org/issue24165

### CODE REVIEW

File: Include/internal/pycore_interp.h [hunk 1]
```diff
@@ -80,6 +80,12 @@ struct _Py_float_state {
     PyFloatObject *free_list;
 };
 
+struct _Py_long_state {
+    // Special free list; same to _Py_float_state but for PyLongObjects.
```
Comment (scoder) [inline Include/internal/pycore_interp.h]:
```suggestion
    // Special free list; similar to _Py_float_state but for PyLongObjects.
```

File: Objects/longobject.c [hunk 2]
```diff
@@ -130,22 +135,34 @@ long_normalize(PyLongObject *v)
 PyLongObject *
 _PyLong_New(Py_ssize_t size)
 {
+    struct _Py_long_state *state = &_PyInterpreterState_GET()->long_state;
+#ifdef Py_DEBUG
+    // _PyLong_New() must not be called after _PyFloat_Fini()
```
Comment (scoder) [inline Objects/longobject.c]: Did you mean `_PyLong_Fini()`?
```suggestion
    // _PyLong_New() must not be called after _PyLong_Fini()
```

**Review by scoder — COMMENTED** [review]
Looks good to me. A few comments need fixing.

File: Misc/NEWS.d/next/Core and Builtins/2020-10-22-19-43-46.bpo-24165.zbVM5Q.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Add free list for long objects.
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2020-10-22-19-43-46.bpo-24165.zbVM5Q.rst]:
```suggestion
Add a free list for long objects.
```

**Comment by pablogsal** [comment]
From the bpo issue:

> Inada-san, you can run pyperformance with this workaround:
> python -m pip install pyperformance==1.0.0
> We are fixing the error soon after https://discuss.python.org/t/pep-641-using-an-underscore-in-the-version-portion-of-python-3-10-compatibility-tags/5513 lands

**Pull request closed without merging**